packages/core/src/payment/strategies/credit-card-redirect/credit-card-redirect-payment-strategy.ts

```diff
@@ -85,11 +90,15 @@ export default class CreditCardRedirectPaymentStrategy extends CreditCardPayment
                 return Promise.reject(error);
             }
 
-            return new Promise(() => this._formPoster.postForm(error.body.three_ds_result.acs_url, {
-                PaReq: error.body.three_ds_result.payer_auth_request || null,
-                TermUrl: error.body.three_ds_result.callback_url || null,
-                MD: error.body.three_ds_result.merchant_data || null,
-            }));
+            return new Promise(() => {
+                if (isRequestError(error)) {
```

Comment:
Few lines above already check if `error` is an instance of `RequestError`.
https://github.com/bigcommerce/checkout-sdk-js/blob/01ba5076cb505de7512d180425a385c5b4c4cc5c/packages/core/src/payment/strategies/credit-card-redirect/credit-card-redirect-payment-strategy.ts#L56

So I think we just need to store the values in a variable outside of the `Promise` closure e.g.:

```ts
const { three_ds_result } = error.body;

return new Promise(() => this._formPoster.postForm(three_ds_result.acs_url, {
    PaReq: three_ds_result.payer_auth_request || null,
    TermUrl: three_ds_result.callback_url || null,
    MD: three_ds_result.merchant_data || null,
}));
```